confmap/provider/httpprovider/provider.go

```diff
@@ -0,0 +1,84 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpprovider // import "go.opentelemetry.io/collector/confmap/provider/httpprovider"
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/provider/internal"
+)
+
+const (
+	schemeName = "http"
+)
+
+type httpClient interface {
+	Get(url string) (resp *http.Response, err error)
+}
```

Comment:
Instead of creating a "client", better test against an http test server, see https://pkg.go.dev/net/http/httptest#NewServer

Reply:
Fixed, that makes sense.

Reply:
Doesn't seem to be fixed

Reply:
Oh, I updated the unit test with httptest package. Do I also need to update provider.go?

Reply:
Fixed!